### PR TITLE
Health indicator

### PR DESCRIPTION
**Pull request by tomerarm**
Health indicators

### CODE REVIEW

File: basenode/src/main/java/io/coti/basenode/controllers/NodeHealthController.java [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+@Slf4j
+@RestController
+@RequestMapping("/health")
+public class NodeHealthController {
+
+    @Autowired
+    IMonitorService monitorService;
+
+    @GetMapping(path = "/state")
```
Comment (gmesika-coti) [inline basenode/src/main/java/io/coti/basenode/controllers/NodeHealthController.java]: change to /total/state

File: nodemanager/src/main/java/io/coti/nodemanager/websocket/WebSocketSender.java [hunk 2]
```diff
@@ -27,5 +30,12 @@ public void notifyNodeDetails(NetworkNodeData networkNodeData, NetworkNodeStatus
                 new NodeMessage(networkNodeData, nodeStatus));
     }
 
+    public void notifyNodeHealthState(Hash nodeHash, BaseNodeMonitorService.HealthState healthState) {
+        log.debug("Node {} is now reported with health state {} ", nodeHash, healthState);
+        messagingSender.convertAndSend("/topic/healthState/nodes/",
```
Comment (gmesika-coti) [inline nodemanager/src/main/java/io/coti/nodemanager/websocket/WebSocketSender.java]: /topic/nodes/health

File: nodemanager/src/main/java/io/coti/nodemanager/websocket/WebSocketSender.java [hunk 3]
```diff
@@ -27,5 +30,12 @@ public void notifyNodeDetails(NetworkNodeData networkNodeData, NetworkNodeStatus
...
+                new NotifyNodeHealthStateChange(nodeHash, healthState));
+        messagingSender.convertAndSend("/topic/healthState/node/" + nodeHash.toString(),
```
Comment (gmesika-coti) [inline nodemanager/src/main/java/io/coti/nodemanager/websocket/WebSocketSender.java]: /topic/node/health

File: basenode/src/main/resources/application.properties [hunk 4]
```diff
@@ -21,4 +21,31 @@ db.restore.source=Remote
 db.restore.hash=
 reset.transactions=false
 native.currency.name=Coti
-native.currency.symbol=COTI
\ No newline at end of file
+native.currency.symbol=COTI
+
+metrics.sample.milisec.interval=6000
```
Comment (gmesika-coti) [inline basenode/src/main/resources/application.properties:26]: remove

File: basenode/src/main/resources/application.properties [hunk 5]
```diff
@@ -21,4 +21,31 @@ db.restore.source=Remote
...
+
+spring.main.allow-circular-references=true
+
+management.endpoint.info.enabled=true
```
Comment (gmesika-coti) [inline basenode/src/main/resources/application.properties:30]: how to limit access to url. maybe just add it to the filter

File: basenode/src/main/java/io/coti/basenode/services/BaseNodeDBRecoveryService.java [hunk 6]
```diff
@@ -82,6 +84,15 @@ public class BaseNodeDBRecoveryService implements IDBRecoveryService {
     private String restoreS3Path;
     private String s3FolderName;
 
+    private BackupInfo lastBackupInfo;
```
Comment (gmesika-coti) [inline basenode/src/main/java/io/coti/basenode/services/BaseNodeDBRecoveryService.java:92]: add /getter and remove methods in below below

File: basenode/src/main/java/io/coti/basenode/http/BaseNodeHealthMetricConstants.java [hunk 7]
```diff
@@ -0,0 +1,47 @@
+package io.coti.basenode.http;
+
+public class BaseNodeHealthMetricConstants {
+
```
Comment (gmesika-coti) [inline basenode/src/main/java/io/coti/basenode/http/BaseNodeHealthMetricConstants.java]: move to constants dir

**Review by gmesika-coti — CHANGES_REQUESTED** [review]

File: nodemanager/src/main/java/io/coti/nodemanager/websocket/WebSocketSender.java (hunk 3 above)
Comment (tomerarm) [reply nodemanager/src/main/java/io/coti/nodemanager/websocket/WebSocketSender.java]: Updated

File: nodemanager/src/main/java/io/coti/nodemanager/websocket/WebSocketSender.java (hunk 2 above)
Comment (tomerarm) [reply nodemanager/src/main/java/io/coti/nodemanager/websocket/WebSocketSender.java]: Updated

File: basenode/src/main/resources/application.properties (hunk 5 above)
Comment (tomerarm) [reply basenode/src/main/resources/application.properties:30]: Updated

File: basenode/src/main/java/io/coti/basenode/http/BaseNodeHealthMetricConstants.java (hunk 7 above)
Comment (tomerarm) [reply basenode/src/main/java/io/coti/basenode/http/BaseNodeHealthMetricConstants.java]: Updated

**Comment by gmesika-coti** [comment]
no need for this PR. we already refactored code.

**Pull request closed without merging**